### PR TITLE
Fixing the type of `SelectItemWithItems` and adding `labelProps`

### LINKED ISSUE #2759: Fixing the type of SelectItemWithItems and adding labelProps
### Which package has an issue?

@yamada-ui/react

### Package Version

1.5.0

### Description

Fixing the type of `SelectItemWithItems` and adding `labelProps`.

```tsx
<OptionGroup
  key={i}
  label={label ?? ""}
  {...(props as HTMLUIProps<"ul">)} // Avoid type conversion here.
>
  {items.map(({ label, value, ...props }, i) => (
    <Option key={i} value={value} {...props}>
      {label}
    </Option>
  ))}
</OptionGroup>
```

Related #2720 
Allow the style to be applied to the label as well, like this issue.

### Link to Reproduction

_No response_

### Steps to Reproduce

2. Click on '...'
3. Scroll down to '...'

### Browser

None

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [ ] iPadOS
- [ ] iOS
- [ ] Android

### Do you know how to fix the issue?

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix?

Yes

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by illionillion**
Closes #2759

### CODE REVIEW

File: packages/components/select/tests/select.test.tsx
Comment (108yen) [inline packages/components/select/tests/select.test.tsx:1]: The purpose of the test is different, so please separate them.

File: packages/components/select/src/use-select.ts
Comment (108yen) [inline packages/components/select/src/use-select.ts:1]: The `labelProps` looks like nothing is being processed, does this need to be put in here?

**Review by 108yen — CHANGES_REQUESTED** [review]

File: packages/components/select/src/use-select.ts
Comment (illionillion) [reply packages/components/select/src/use-select.ts:1]: Thanks I fixed

**Comment by illionillion** [comment]
@108yen I fixed. Please review again.

File: packages/components/select/tests/select.test.tsx [hunk 1]
```diff
@@ -117,6 +117,25 @@ describe("<Select />", () => {
...
+        </Select>,
+      )
+
+      const input = await screen.findByRole("combobox")
+
+      await user.click(input)
+
+      await expect(screen.findByText("Numbers")).resolves.toHaveStyle({
+        fontSize: "12px",
+      })
```
Comment (108yen) [inline packages/components/select/tests/select.test.tsx:136]: `await expect` probably doesn't make sense.
Please fix.
```tsx
const group = await screen.findByText("Numbers")
      expect(group).toHaveStyle({
        fontSize: "12px",
      })
```

File: packages/components/select/tests/select.test.tsx
Comment (108yen) [inline packages/components/select/tests/select.test.tsx:1]: And while you're at it, can you include `should be displayed properly when NOT present` in the `header rendering`?

**Review by 108yen — CHANGES_REQUESTED** [review]

File: packages/components/select/tests/select.test.tsx (hunk 1 above)
Comment (illionillion) [reply packages/components/select/tests/select.test.tsx]: fixed

File: packages/components/select/tests/select.test.tsx
Comment (illionillion) [reply packages/components/select/tests/select.test.tsx:1]: Are you suggesting that we add a test to see if it works correctly even if we don't specify anything for the header property of Select?

Comment (108yen) [reply packages/components/select/tests/select.test.tsx:1]: @illionillion 
Please make this change!

```diff
  describe("header rendering", () => {
    test("should be displayed properly when present.", () => {
      const HEADER_TEXT = "Header"
      const HEADER_STYLE = { background: "green" }

      const renderHeader = () => {
        return <header style={HEADER_STYLE}>{HEADER_TEXT}</header>
      }

      render(
        <Select
          header={renderHeader()}
          placeholder="Select numbers"
          placeholderInOptions={true}
          items={[
            { label: "One", value: "" },
            { label: "Two", value: "" },
          ]}
        />,
      )

      const headerElement = screen.getByText(HEADER_TEXT)

      expect(headerElement).toBeInTheDocument()
      expect(headerElement).toHaveStyle(HEADER_STYLE)
      expect(headerElement.parentElement).toHaveClass("ui-select__header")
    })
-  })

  test("should be displayed properly when NOT present", () => {
    const { container } = render(
      <Select
        placeholder="Select numbers"
        placeholderInOptions={true}
        items={[
          { label: "One", value: "" },
          { label: "Two", value: "" },
        ]}
      />,
    )

    const headerElement = container.querySelector(".ui-select__header")
    expect(headerElement).toBeNull()
  })
+  })
```

Comment (illionillion) [reply packages/components/select/tests/select.test.tsx:1]: Thanks I fixed it.

**Comment by illionillion** [comment]
Please review again. @108yen

File: packages/components/select/tests/select.test.tsx
Comment (108yen) [reply packages/components/select/tests/select.test.tsx:1]: @illionillion 
Wrong.

Comment (illionillion) [reply packages/components/select/tests/select.test.tsx:1]: Sorry, I fixed.
deleted duplicates.

**Review by 108yen — APPROVED** [review]
Thank you!

File: packages/components/select/src/use-select.ts [hunk 2]
```diff
@@ -792,6 +792,10 @@ export interface UseSelectOptionGroupProps extends HTMLUIProps<"ul"> {
    * The label of the option group.
    */
   label: string
+  /**
+   * Props for option group element.
+   */
+  labelProps?: HTMLUIProps<"span">
```
Comment (hirotomoyamada) [inline packages/components/select/src/use-select.ts:798]: Do not define props to customize components here.  Set to the `option-group.tsx` side.

File: packages/components/select/src/select.tsx [hunk 3]
```diff
@@ -34,7 +34,7 @@ interface SelectItemWithValue extends OptionProps {
 
 interface SelectItemWithItems extends OptionGroupProps {
   items?: SelectItemWithValue[]
-}
+} & UseSelectOptionGroupProps
```
Comment (hirotomoyamada) [inline packages/components/select/src/select.tsx:37]: This fix is no longer needed in #2803. Please remove this modification.

File: .changeset/sweet-shoes-lie.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/select": patch
+---
+
+Fixing the type of SelectItemWithItems and adding labelProps.
```
Comment (hirotomoyamada) [inline .changeset/sweet-shoes-lie.md:5]: Please correct this description as well.

https://github.com/yamada-ui/yamada-ui/pull/2760/files#r1762265693

File: packages/components/select/src/use-select.ts (hunk 2 above)
Comment (illionillion) [reply packages/components/select/src/use-select.ts]: fixed

File: packages/components/select/src/select.tsx (hunk 3 above)
Comment (illionillion) [reply packages/components/select/src/select.tsx]: fixed

File: .changeset/sweet-shoes-lie.md (hunk 4 above)
Comment (illionillion) [reply .changeset/sweet-shoes-lie.md]: fixed

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**